Shared/Server Protocol/CodableError.swift

```diff
@@ -0,0 +1,241 @@
+//
+//  CodableError.swift
+//  AltKit
+//
+//  Created by Riley Testut on 3/5/20.
+//  Copyright © 2020 Riley Testut. All rights reserved.
+//
+
+import Foundation
+
+// Can only automatically conform ALTServerError.Code to Codable, not ALTServerError itself
+extension ALTServerError.Code: Codable {}
+
+private extension ErrorUserInfoKey
+{
+    static let altLocalizedDescription: String = "ALTLocalizedDescription"
+    static let altLocalizedFailureReason: String = "ALTLocalizedFailureReason"
+    static let altLocalizedRecoverySuggestion: String = "ALTLocalizedRecoverySuggestion"
+    static let altDebugDescription: String = "ALTDebugDescription"
+}
+
+extension CodableError
+{
+    enum UserInfoValue: Codable
+    {
+        case unknown
+        case string(String)
+        case number(Int)
+        case error(NSError)
+        case codableError(CodableError)
+        indirect case array([UserInfoValue])
+        indirect case dictionary([String: UserInfoValue])
+        
+        var value: Any? {
+            switch self
+            {
+            case .unknown: return nil
+            case .string(let string): return string
+            case .number(let number): return number
+            case .error(let error): return error
+            case .codableError(let error): return error.error
+            case .array(let array): return array
```

Comment:
BUG: We need to return the wrapped values for .array and .dictionary, not the UserInfoValue's directly